java-lib/src/main/java/com/wavefront/ingester/GraphiteDecoder.java

```diff
@@ -26,19 +27,24 @@
       .appendOptionalTimestamp().whiteSpace()
       .appendAnnotationsConsumer().whiteSpace().build();
   private final String hostName;
+  private LinkedHashSet<String> customSourceTags;
 
-  public GraphiteDecoder() {
+  public GraphiteDecoder(LinkedHashSet<String> customSourceTags) {
```

Comment:
Typically, the idea is to make inputs as generic as possible (since it's perfectly possible that there are other ordered set implementations). I think in your case, you can actually use Collection<String> even and state that the iteration order is the order in which the tag would be considered. You can then even use a list (since you really want an error if someone puts a tag twice anyways in the config/arg).